### PR TITLE
First push with draft changes

### PR DESCRIPTION
**Pull request by peet-j**

### CODE REVIEW

File: src/src/main/java/com/example/actuatorservice/Greeting.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		this.content = content;
+	}
+
+	public long getId() {
+		return id;
+	}
+
+	public String getContent() {
+		return content;
+	}
```
Comment (ElvenSpellmaker) [inline src/src/main/java/com/example/actuatorservice/Greeting.java:19]: This file is in tabs, probably want to settle on a single type?

File: src/src/main/java/com/example/actuatorservice/HelloWorldApplication.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class HelloWorldApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(HelloWorldApplication.class, args);
+    }
+
+}
```
Comment (ElvenSpellmaker) [inline src/src/main/java/com/example/actuatorservice/HelloWorldApplication.java:13]: We should always have newlines on the ends of files, lots of files here don't, this is just one example.

File: src/test/java/com/xxAMIDOxx/xxSTACKSxx/ApplicationTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Test
+    public void shouldReturn200WhenSendingRequestToManagementEndpoint() {
+        ResponseEntity<Map> entity = this.testRestTemplate.getForEntity(
+                "http://localhost:" + this.mgt + "/actuator/info", Map.class);
+
+        then(entity.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+}
```
Comment (ElvenSpellmaker) [inline src/test/java/com/xxAMIDOxx/xxSTACKSxx/ApplicationTest.java:46]: Newline please

File: src/main/java/com/xxAMIDOxx/xxSTACKSxx/Application.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class Application {
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+}
```
Comment (ElvenSpellmaker) [inline src/main/java/com/xxAMIDOxx/xxSTACKSxx/Application.java:13]: Newline please

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (ElvenSpellmaker) [inline pom.xml:82]: Newline please

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }
+}
+
+springBoot {
+    mainClassName = 'com.xxAMIDOxx.xxSTACKSxx.Application'
+}
+
+test {
+    useJUnitPlatform()
+}
```
Comment (ElvenSpellmaker) [inline build.gradle:35]: Newline please

**Comment by ElvenSpellmaker** [comment]
Could we have a `.gitattributes` file with the contents:
```
* text eol=lf
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,18 @@
...
+
+> ./gradlew bootRun
+
+## Build Docker Image
+
+> docker build -t <image-tag> .
+
+## Swagger/OAS
+
+Automatically generated. Go to: http://localhost:9000/api-docs
```
Comment (ElvenSpellmaker) [inline README.md:14]: Can we change this endpoint to match what .NET stacks has?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,18 @@
...
+
+## Health check
+
+Available at: http://localhost:9001/actuator/health
```
Comment (ElvenSpellmaker) [inline README.md:18]: Likewise can we match .NET for health endpoint?

Comment (peet-j) [reply README.md:18]: We can do this. `/health` is what is being used. Easy config change

File: README.md (hunk 7 above)
Comment (peet-j) [reply README.md:14]: .NET is using `/swagger/index.html` - though not sure I like it. It's easy to reconfigure the path though

Comment (peet-j) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (peet-j) [reply README.md]: done

File: src/test/java/com/xxAMIDOxx/xxSTACKSxx/ActuatorTest.java [hunk 9]
```diff
@@ -17,30 +14,31 @@
...
+public class ActuatorTest {
 
     @Value("${local.management.port}")
     private int mgt;
 
     @Autowired
     private TestRestTemplate testRestTemplate;
 
+    private String getBaseURL(final int port) {
+        return "http://localhost:" + port;
```
[the remark this answers is not in the document]
Comment (peet-j) [reply src/test/java/com/xxAMIDOxx/xxSTACKSxx/ActuatorTest.java:26]: Integration test. Spins up the API locally and performs http calls against it for testing.

**Comment by peet-j** [comment]
> Could we have a `.gitattributes` file with the contents:
> 
> ```
> * text eol=lf
> ```

Add 👍

File: src/main/java/com/xxAMIDOxx/xxSTACKSxx/Application.java (hunk 4 above)
Comment (ElvenSpellmaker) [inline src/main/java/com/xxAMIDOxx/xxSTACKSxx/Application.java]: newline please =)

File: pom.xml (hunk 5 above)
Comment (ElvenSpellmaker) [inline pom.xml]: newline please =)

File: build.gradle (hunk 6 above)
Comment (ElvenSpellmaker) [inline build.gradle]: newline please =)

File: src/test/java/com/xxAMIDOxx/xxSTACKSxx/ApplicationTest.java (hunk 3 above)
Comment (ElvenSpellmaker) [inline src/test/java/com/xxAMIDOxx/xxSTACKSxx/ApplicationTest.java]: newline please =)

**Comment by ElvenSpellmaker** [comment]
Once the Sonar build passes we should merge this.

**Comment by ElvenSpellmaker** [comment]
:shipit:

**Pull request merged**